src/Bicep.Core/Extensions/IPositionableExtensions.cs

```diff
@@ -8,9 +8,18 @@ namespace Bicep.Core.Extensions
     public static class IPositionableExtensions
     {
         public static TextSpan ToZeroLengthSpan(this IPositionable positionable)
-            => new TextSpan(positionable.Span.Position, 0);
+            => new(positionable.Span.Position, 0);
+
+        public static int GetPosition(this IPositionable positionable)
+            => positionable.Span.Position;
 
         public static int GetEndPosition(this IPositionable positionable)
             => positionable.Span.Position + positionable.Span.Length;
+
+        public static bool IsOverlapping(this IPositionable positionable, int position)
+            => positionable.GetPosition() <= position && position <= positionable.GetEndPosition();
+
+        public static bool IsEnclosing(this IPositionable positionable, int position)
+            => positionable.GetPosition() < position && position < positionable.GetEndPosition();
```

Comment:
>< [](http://example.com/codeflow?start=42&length=1)

Is it intentional to return false when `position == positionable.GetPosition()`? I think all other checks have an inclusive version of that check.

Reply:
If it can be inclusive, then we do have the `Contains` method on `TextSpan`.